keps/sig-windows/2578-windows-conformance/README.md

```diff
@@ -0,0 +1,474 @@
+
+One way to implement such a tool is to, specifically, target all available "ginkgo" tests in Kubernetes core itself, and run these one at a time (or concurrently).  Not we *do not* assert that these tests *must* run in parallel in order to comprise a conformance cluster (although this is often done in CI, its generally not a requirement of Kubernetes Conformance for linux clusters, either, since cluster capacity is not related to the functionality of a cluster). As an example, one such implementation of such a Conformance tool might be:
+
+```
+--ginkgo.focus=\\[Conformance\\]|\\[NodeConformance\\]|\\[sig-windows\\]|\\[sig-apps\\].CronJob --ginkgo.skip=\\[LinuxOnly\\]|\\[k8s.io\\].Pods.*should.cap.back-off.at.MaxContainerBackOff.\\\[NodeConformance\\]|\\[k8s.io\\].Pods.*should.have.their.auto-restart.back-off.timer.reset.on.image.update.\\[Slow\\]\\[NodeConformance\\]"
+```
+
+Currently, the de facto standard for Windows testing resides at https://github.com/kubernetes-sigs/windows-testing, and we suggest this tag to users, however it should be noted that this does *not* conform to the specification in this document:
+
+- it triggers some tests which uses non `pause` images for validating functionality.
+- it doesn't implement the pod churn or `emptyDir` sharing tests.
+- it doesn't implement the *reboot* tests which are defined in this specification.
+
+The ability to specify HostProcess, NetworkPolicy, and other tests in this document would follow a similar tagging strategy, and any such tool could just invoke the existing Kubernetes end-to-end test binary with appropriate `focus` and `skip` arguments.
+
+#### Option 2 - Sonobuoy implementation for convenience
+
+Given the complexity of the above tagging scheme, we propose a concrete example of a binary program for Windows verification. 
+This was the option chosen for this KEP implementation.
+
+```
+sonobuoy run --mode=windows-conformance --windows.hostProcess --windows.activeDirectory ...
+# which prints out the exact tests so that the actual test coverage is clear
+```
+
+In the above invocation, we verify
+
+- Core Conformance
+- HostProcess
+- ActiveDirectory 
+
+functionality for a specific Kubernetes release. The resulting output would be obtained using
+
+```
+sonobuoy status
+```
+
+Which would report a binary result of `Pass` or `Fail`. This tool could thus be used to establish a rigorous standard for portable verification of Kubernetes Windows based products across Vendors for both internal and external diagnostic purposes.
+
+
+##### Implementation details for Sonobuoy
+
+Below is an example of running the Windows Operational Readiness conformance tests with Sonobuoy
+
+```
+sonobuoy plugin install windows-operational-readiness <url>
+sonobuoy run -p windows-operational-readiness
+```
+
+The URL can point to any custom URL. We have published a few examples of Windows e2e tests here: https://github.com/vmware-tanzu/sonobuoy-plugins/tree/master/windows-e2e
```

Comment:
where will this canonical list live?  Since the test definitions are in these configurations they could run any set of tests configured and call it good.

Reply:
updated to point to the GA section as the final place we put the endpoint for all these artifacts.